index.js

```diff
@@ -0,0 +1,270 @@
+        document.querySelector('input').value = string
+
+    }
+    clear() {
+        string = " ";
+        document.querySelector('input').value = string
+    }
+    delete() {
+        string = string.slice(0, -1);
+        document.querySelector('input').value = string
+    }
+    absolute() {
+        string = Math.abs(string)
+        document.querySelector('input').value = string
+    }
+    exponential() {
+        // string = Math.exp(string)
+        let lastNum = this.getLastNumber(document.querySelector('input').value); // Use 'this' to call the method on the instance
+
+        if (lastNum === "") {
+            document.querySelector('input').value = 'Error: Invalid Log';
+            return;
+        }
+        let result = Math.exp(lastNum);
+        let tempInput = document.querySelector('input').value.slice(0, document.querySelector('input').value.length - lastNum.length);
+        document.querySelector('input').value = tempInput + result;
+    }
+    inverse() {
+        string = 1 / string;
+        let lastNum = this.getLastNumber(document.querySelector('input').value); // Use 'this' to call the method on the instance
+
+        if (lastNum === "") {
+            document.querySelector('input').value = 'Error: Invalid Log';
+            return;
+        }
+        let result = 1 / lastNum;
+        let tempInput = document.querySelector('input').value.slice(0, document.querySelector('input').value.length - lastNum.length);
+        document.querySelector('input').value = tempInput + result;
+    }
+    square() {
+        // string = Math.pow(string, 2);
+
+        let lastNum = this.getLastNumber(document.querySelector('input').value); // Use 'this' to call the method on the instance
+
+        if (lastNum === "") {
+            document.querySelector('input').value = 'Error: Invalid Log';
+            return;
+        }
+        let result = Math.pow(lastNum, 2);
+        let tempInput = document.querySelector('input').value.slice(0, document.querySelector('input').value.length - lastNum.length);
+        document.querySelector('input').value = tempInput + result;
+    }
+    squareRoot() {
+
+        let lastNum = this.getLastNumber(document.querySelector('input').value); // Use 'this' to call the method on the instance
+
+        if (lastNum === "") {
+            document.querySelector('input').value = 'Error: Invalid Log';
+            return;
+        }
+        let result = Math.sqrt(lastNum);
+        let tempInput = document.querySelector('input').value.slice(0, document.querySelector('input').value.length - lastNum.length);
+        document.querySelector('input').value = tempInput + result;
+    }
+    factorial() {
+        let lastNum = this.getLastNumber(document.querySelector('input').value); // Use 'this' to call the method on the instance
+
+        if (lastNum === "") {
+            document.querySelector('input').value = 'Error: Invalid Input';
```

Comment:
```suggestion
const input = document.....
            input.value = 'Error: Invalid Input';
```

create constant for queryselector.